packages/cloudflare/src/cli/build/patches/plugins/require-page.ts

```diff
@@ -20,27 +19,37 @@ export function inlineRequirePagePlugin(buildOpts: BuildOptions) {
         async ({ path }) => {
           const jsCode = await readFile(path, "utf8");
           if (/function requirePage\(/.test(jsCode)) {
-            return { contents: patchCode(jsCode, getRule(buildOpts)) };
+            return { contents: patchCode(jsCode, await getRule(buildOpts)) };
           }
         }
       );
     },
   };
 }
 
-function getRule(buildOpts: BuildOptions) {
+async function getRule(buildOpts: BuildOptions) {
   const { outputDir } = buildOpts;
   const serverDir = join(outputDir, "server-functions/default", getPackagePath(buildOpts), ".next/server");
 
   const pagesManifestFile = join(serverDir, "pages-manifest.json");
   const appPathsManifestFile = join(serverDir, "app-paths-manifest.json");
 
-  const pagesManifests: string[] = existsSync(pagesManifestFile)
-    ? Object.values(JSON.parse(readFileSync(pagesManifestFile, "utf-8")))
-    : [];
-  const appPathsManifests: string[] = existsSync(appPathsManifestFile)
-    ? Object.values(JSON.parse(readFileSync(appPathsManifestFile, "utf-8")))
-    : [];
+  let pagesManifests: string[] = [];
+  try {
+    pagesManifests = Object.values(JSON.parse(await readFile(pagesManifestFile, "utf-8")));
+  } catch {
+    // The file does not exists
+    pagesManifests = [];
+  }
```

Comment:
just wanted to point out that this (and the one below) `catch` block catches any exception, so it's not just if the file doesn't exist but for example if the file is malformed (and `JSON.parse` throws), I think that's fine here, but the `// The file does not exists` might be a bit off/misleading in my opinion

Reply:
That's right but if this file is broken, the app would be broken anyway